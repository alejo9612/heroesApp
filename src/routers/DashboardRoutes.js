import React from 'react';
import { Redirect, Route, Switch } from 'react-router';
import { DcPage } from '../components/dc/DcPage';
import { HeroesPage } from '../components/heroes/HeroesPage';
import { MarvelPage } from '../components/marvel/MarvelPage';
import { SearchPage } from '../components/search/SearchPage';
import { Navbar } from '../components/ui/Navbar';

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar/>

            <div className="container">
                <Switch>
                    <Route exact path="/marvel" component={MarvelPage}/>
                    <Route exact path="/heroe/:heroeId" component={HeroesPage}/>
                    <Route exact path="/dc" component={DcPage}/>
                    <Route exact path="/Buscar" component={SearchPage}/>
                    <Redirect to="/marvel"/>
                </Switch>
            </div>
        </>
    )
}
