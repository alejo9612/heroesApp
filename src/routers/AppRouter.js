import React, { useContext } from 'react';
import {
    BrowserRouter as Router,
    Switch
} from "react-router-dom";
import { AuthContext } from '../auth/AuthContext'; 
import { DashboardRoutes } from './DashboardRoutes';
import { LoginPage } from '../components/login/LoginPage';
import { PrivateRouter } from './PrivateRouter';
import { PublicRoute } from './PublicRoute';

export const AppRouter = () => {

    const {user} = useContext(AuthContext);


    return (
        <Router>
            <div>
                <Switch>
                    <PublicRoute exact path="/login" 
                                 component={LoginPage}
                                //  isAuthenticated={user.logged}
                    />
                    <PrivateRouter path="/" 
                                   component={DashboardRoutes}
                                   isAuthenticated={user.logged}
                    />
                </Switch>
            </div>
        </Router>
    )
}
