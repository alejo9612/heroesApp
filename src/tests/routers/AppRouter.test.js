import React from "react";
import { mount } from "enzyme";
import { AppRouter } from '../../routers/AppRouter';
import { AuthContext } from "../../auth/AuthContext";


describe('Se levarán a cabo lass ruebass del AppRouter', () => {

    //Creamos la simulación del context para que se pueda llevar a cabo la prueba
    const contextValue = {
        dispatch : jest.fn(),
        user: {
            logged: false
        }
    }

    test('Debe de mostrar el login si no se encuentra autenticado', () => {

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        )

        // console.log(wrapper.html());
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar el componente de marvel si está autenticado', () => {

        const contextValue = {
            dispatch : jest.fn(),
            user: {
                logged: true,
                name:'Carnaval'
            }
        }
       
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        )

        //console.log(wrapper.html());
        expect(wrapper.find('.container').exists()).toBe(true);
    });

});
