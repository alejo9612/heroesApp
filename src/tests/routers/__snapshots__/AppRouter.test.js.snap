// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Se levar√°n a cabo lass ruebass del AppRouter Debe de mostrar el login si no se encuentra autenticado 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      className="btn btn-success"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
